src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx

```diff
@@ -10,57 +9,52 @@ import * as ReportUtils from '@libs/ReportUtils';
 import * as Download from '@userActions/Download';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps as anchorForAttachmentsOnlyDefaultProps, propTypes as anchorForAttachmentsOnlyPropTypes} from './anchorForAttachmentsOnlyPropTypes';
-
-const propTypes = {
-    /** Press in handler for the link */
-    onPressIn: PropTypes.func,
-
-    /** Press out handler for the link */
-    onPressOut: PropTypes.func,
+import {Download as OnyxDownload} from '@src/types/onyx';
+import AnchorForAttachmentsOnlyProps from './types';
 
+type BaseAnchorForAttachmentsOnlyOnyxProps = {
     /** If a file download is happening */
-    download: PropTypes.shape({
-        isDownloading: PropTypes.bool.isRequired,
-    }),
-
-    ...anchorForAttachmentsOnlyPropTypes,
+    download: OnyxEntry<OnyxDownload>;
 };
 
-const defaultProps = {
-    onPressIn: undefined,
-    onPressOut: undefined,
-    download: {isDownloading: false},
-    ...anchorForAttachmentsOnlyDefaultProps,
-};
+type BaseAnchorForAttachmentsOnlyProps = AnchorForAttachmentsOnlyProps &
+    BaseAnchorForAttachmentsOnlyOnyxProps & {
+        /** Press in handler for the link */
+        onPressIn?: () => void;
+
+        /** Press out handler for the link */
+        onPressOut?: () => void;
+    };
 
-function BaseAnchorForAttachmentsOnly(props) {
-    const sourceURL = props.source;
+function BaseAnchorForAttachmentsOnly({style, source = '', displayName = '', download, onPressIn, onPressOut}: BaseAnchorForAttachmentsOnlyProps) {
+    const sourceURL = source;
```

Comment:
NAB: Do we really need this? I think it was used to not repeat `props.source` over and over

Reply:
It is done like that for a better default initialisation of displayName and source props.
I find it easier to identify the default value by destructuring the props.